### PR TITLE
Move foundation examples to Components folder

### PR DESCRIPTION
**Pull request by precious-onyenaucheya-ons**
### What is the context of this PR?

Fixes https://github.com/ONSdigital/design-team/issues/52
It relocates the examples of list, quote, text-indent, and address output from the foundations folder to the component folder. This change ensures that the macro options file and examples file are co-located for easier accessibility. Additionally, the grid examples have been moved and a new folder has been created for them.

### How to review this PR

Verify that all tests and visual tests have been updated correctly.

### Checklist

This needs to be completed by the person raising the PR.

-   [x] I have selected the correct Assignee
-   [x] I have linked the correct Issue

### CODE REVIEW

File: src/scss/main.scss [hunk 1]
```diff
@@ -27,7 +27,7 @@
 @import '../components/fieldset/fieldset';
 @import '../components/footer/footer';
 @import '../components/header/header';
-@import '../components/helpers/grid';
+@import '../components/helpers/grid/grid';
```
Comment (rmccar) [inline src/scss/main.scss:30]: I think we should either remove the helpers folder or move the grid files into foundations. Grid isn't really a component

File: backstop_data/bitmaps_reference/ds-vr-test__components_helpers_grid_example-grid-8_0_document_0_desktop.png
Comment (rmccar) [inline backstop_data/bitmaps_reference/ds-vr-test__components_helpers_grid_example-grid-8_0_document_0_desktop.png:1]: This grid examples are broken

**Review by rmccar — CHANGES_REQUESTED** [review]

File: src/scss/main.scss (hunk 1 above)
Comment (alessioventuriniAND) [reply src/scss/main.scss:30]: I agree I think one should go back to foundation as it's not a component. Reason I am saying this is not a component is that we don't expect users to use this grid macro but the grid classes.

**Review by alessioventuriniAND — COMMENTED** [review]
Could you also run the reference command for the visual regression tests? It will remove all the old references that are not longer needed like all the screenshots of the same component but with the old name.

File: backstop_data/bitmaps_reference/ds-vr-test__components_helpers_grid_example-grid-8_0_document_0_desktop.png
Comment (precious-onyenaucheya-ons) [reply backstop_data/bitmaps_reference/ds-vr-test__components_helpers_grid_example-grid-8_0_document_0_desktop.png:1]: i have fixed the issue

**Review by rmccar — APPROVED** [review]

**Review by SriHV — APPROVED** [review]

**Pull request merged**